flow/project.py

```diff
@@ -2082,7 +2079,7 @@ def _get_operations_status(self, jobs, cached_status):
 
         groups = [self._groups[name] for name in self.operations]
         for group in groups:
-            if get_aggregate_id(jobs) in self._get_aggregate_store(group.name):
+            if get_aggregate_id(jobs) in self._group_to_aggregator[group]:
```

Comment:
this and line 2186 are where we get the performance boost, yes?

Reply:
That is a small speedup, yes. The bigger performance boost is from replacing code like:
```python
for group in groups:
    for aggregate in self._get_aggregate_store(group.name).values()
        ...
```
with the selection logic that is partially implemented in this PR and more thoroughly implemented in #416.